src/main/java/org/repodriller/scm/GitRepository.java

```diff
@@ -316,11 +299,57 @@ public Commit getCommit(String id) {
 		return mappedBranches;
 	}
 
+	private Modification getModificationFromDiff(Repository repo, DiffEntry diff) throws IOException {
+		ModificationType change = Enum.valueOf(ModificationType.class, diff.getChangeType().toString());
+
+		String oldPath = diff.getOldPath();
+		String newPath = diff.getNewPath();
+
+		String diffText = "";
+		String sc = "";
+		if (diff.getChangeType() != ChangeType.DELETE) {
+			diffText = getDiffText(repo, diff);
+			sc = getSourceCode(repo, diff);
+		}
+
+		if (diffText.length() > maxSizeOfDiff) {
+			log.error("diff for " + newPath + " too big");
+			diffText = "-- TOO BIG --";
+		}
+
+		return new Modification(oldPath, newPath, change, diffText, sc);
+	}
+
 	private List<DiffEntry> diffsForTheCommit(Repository repo, RevCommit commit) throws IOException {
 
 		AnyObjectId currentCommit = repo.resolve(commit.getName());
 		AnyObjectId parentCommit = commit.getParentCount() > 0 ? repo.resolve(commit.getParent(0).getName()) : null;
 
+		return this.getDiffBetweenCommits(repo, parentCommit, currentCommit);
+	}
+
+	@Override
+	public List<Modification> getDiffBetweenCommits(String priorCommitHash, String laterCommitHash) {
+		try (Git git = openRepository()) {
+			Repository repo = git.getRepository();
+			AnyObjectId priorCommit = repo.resolve(priorCommitHash);
+			AnyObjectId laterCommit = repo.resolve(laterCommitHash);
+
+			List<DiffEntry> diffs = this.getDiffBetweenCommits(repo, priorCommit, laterCommit);
+			List<Modification> modifications = new ArrayList<Modification>();
+			for (DiffEntry diff : diffs) {
+				Modification m = this.getModificationFromDiff(repo, diff);
```

Comment:
Dare I suggest we rename this method to diffToModification, or even further, add a constructor on Modification that handles this for us?

Reply:
Sure to the first, dunno about the second. @mauricioaniche?